syntax = "proto3";

package todo;

service TodoService {
  rpc CreateTodo (CreateTodoRequest) returns (TodoResponse);
  rpc GetTodo (GetTodoRequest) returns (TodoResponse);
}

message CreateTodoRequest {
  string title = 1;
  string description = 2;
  int32 user_id = 3; // validated with UserService
}

message GetTodoRequest {
  int32 id = 1;
}

message TodoResponse {
  int32 id = 1;
  string title = 2;
  string description = 3;
  int32 user_id = 4;
}
